### ISSUE TITLE
Bug: False positive for ChainingConstructorIgnoresParameter for array concatenation passing

### ISSUE THREAD

**Issue by Fox2Code** [google/error-prone#5472]
The following pattern:
```java
public class MyClass {
        final String[] targets;

	MyClass(String target) {
		this.targets = new String[]{target};
	}

	MyClass(String[] targets) {
		this.targets = targets;
	}

	MyClass(String target, String... targets) {
		String[] newTargets = new String[targets.length + 1];
		newTargets[0] = target;
		System.arraycopy(targets, 0, newTargets, 1, targets.length);
		this(newTargets);
	}
}
```
Cause a [ChainingConstructorIgnoresParameter](https://errorprone.info/bugpattern/ChainingConstructorIgnoresParameter) false-positive!

Found out while doing stuff on my project, original code is there:
https://github.com/Fox2Code/Hypertale/blob/master/patcher/src/main/java/com/fox2code/hypertale/patcher/patches/HypertalePatch.java

**Comment by cpovirk** [comment]
Sorry about that.

For what it's worth, I think you might be able to avoid it by renaming some constructor parameters, perhaps either...

```java
MyClass(String onlyTarget)
```

...or...

```java
MyClass(String firstTarget, String... otherTargets)
```

...kind of like [this example](https://guava.dev/releases/snapshot-jre/api/docs/com/google/common/collect/Sets.html#immutableEnumSet(E,E...)) in the latter case.

This is the first such report I can recall at the moment, so my guess is that we won't get around to fixing it unless we hear more reports, but thank you for taking the time to report it in any case.
